am-lib/src/test/java/uk/gov/hmcts/reform/amlib/PermissionTest.java

```diff
@@ -0,0 +1,136 @@
+package uk.gov.hmcts.reform.amlib;
+
+import org.junit.Test;
+import uk.gov.hmcts.reform.amlib.enums.Permission;
+import uk.gov.hmcts.reform.amlib.exceptions.UnsupportedPermissionsException;
+import uk.gov.hmcts.reform.amlib.utils.Permissions;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatExceptionOfType;
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+@SuppressWarnings("PMD")
+public class PermissionTest {
+
+    @Test
+    public void sumOf_whenPassingPermissions_theSumOfValuesIsCalculated() {
+        Set<Permission> permissions = new HashSet<>(Arrays.asList(Permission.CREATE, Permission.READ));
+
+        int sum = Permissions.sumOf(permissions);
+
+        int expectedSum = Permission.CREATE.getValue() + Permission.READ.getValue();
+        assertThat(sum).isEqualTo(expectedSum);
+    }
+
+    @Test
+    public void sumOf_whenPassingNoPermissions_theSumOfValuesIsZero() {
+        Set<Permission> permissions = new HashSet<>();
+
+        int sum = Permissions.sumOf(permissions);
+
+        assertThat(sum).isEqualTo(0);
+    }
+
+    @Test
+    public void buildPermissions_sumOfPermissionsValueOne_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.buildPermissions(1);
+
+        assertThat(permissions).containsOnly(Permission.CREATE);
+    }
+
+    @Test
+    public void buildPermissions_sumOfPermissionsValueThree_ExpectArray() throws UnsupportedPermissionsException {
+        Set<Permission> permissions = Permissions.buildPermissions(3);
```

Comment:
I do not quite like how it reads. IMO something like `Permissions.fromSumOf(3)` reads better.

Reply:
👍